packages/framework/src/Twig/RequiredSettingExtension.php

```diff
@@ -0,0 +1,235 @@
+        protected readonly CountryFacade $countryFacade,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('render_required_settings', $this->renderRequiredSettings(...), ['is_safe' => ['html']]),
+        ];
+    }
+
+    /**
+     * @return string|null
+     */
+    public function renderRequiredSettings(): ?string
+    {
+        $requiredSettingsMessages = [];
+
+        $this->checkEnabledMailTemplatesHaveTheirBodyAndSubjectFilled($requiredSettingsMessages);
+        $this->checkAtLeastOneUnitExists($requiredSettingsMessages);
+        $this->checkDefaultUnitIsSet($requiredSettingsMessages);
+        $this->checkAtLeastOneWarehouseExists($requiredSettingsMessages);
+        $this->checkAtLeastOneCountryExists($requiredSettingsMessages);
+        $this->checkMandatoryArticlesExist($requiredSettingsMessages);
+        $this->checkAllSliderNumericValuesAreSet($requiredSettingsMessages);
+
+        if (count($requiredSettingsMessages) === 0) {
+            return null;
+        }
+
+        return $this->twig->render(
+            '@ShopsysFramework/Components/RequiredSettings/requiredSettings.html.twig',
+            [
+                'requiredSettingsMessages' => $requiredSettingsMessages,
+            ],
+        );
+    }
+
+    /**
+     * @param string[] $warnings
+     */
+    protected function checkEnabledMailTemplatesHaveTheirBodyAndSubjectFilled(array &$warnings): void
+    {
+        if ($this->mailTemplateFacade->existsTemplateWithEnabledSendingHavingEmptyBodyOrSubject()) {
+            $warnings[] = t(
+                '<a href="%url%">Some required email templates are not fully set.</a>',
+                [
+                    '%url%' => $this->router->generate('admin_mail_template'),
+                ],
+            );
+        }
+    }
+
+    /**
+     * @param string[] $warnings
+     */
+    protected function checkAtLeastOneUnitExists(array &$warnings): void
+    {
+        if ($this->unitFacade->getCount() === 0) {
+            $warnings[] = t(
+                '<a href="%url%">There are no units, you need to create some.</a>',
+                [
+                    '%url%' => $this->router->generate('admin_unit_list'),
+                ],
+            );
+        }
+    }
+
+    /**
+     * @param string[] $warnings
+     */
+    protected function checkDefaultUnitIsSet(array &$warnings): void
+    {
+        try {
+            $this->unitFacade->getDefaultUnit();
+        } catch (UnitNotFoundException) {
+            $warnings[] = t(
+                '<a href="%url%">Default unit is not set.</a>',
+                [
+                    '%url%' => $this->router->generate('admin_unit_list'),
+                ],
+            );
+        }
+    }
+
+    /**
+     * @param string[] $warnings
+     */
+    protected function checkAtLeastOneWarehouseExists(array &$warnings): void
```

Comment:
in the code itself, I would rather use "stock" instead of warehouse, but I am not sure whether this rule is strictly followed even now, so :shrug:

Reply:
fixed in c3bbb989a3